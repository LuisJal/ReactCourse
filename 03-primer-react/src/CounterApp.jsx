import React from "react";
import PropTypes from "prop-types";
import { useState } from "react";

export const CounterApp = ({ value }) => {
  const [counter, setCounter] = useState(value);

  //Sumar
  const handleAdd = () => setCounter((c) => c + 1);
  //Restar
  const handleSubtract = () => setCounter((c) => c - 1);
  //Reiniar
  const handleReset = () => setCounter((c) => (c = 0));

  return (
    <>
      <h1>CounterAppLuis</h1>
      <h2 data-testid="contador">{counter}</h2>
      <div className="botones">
        <button onClick={handleAdd}>+1</button>
        <button onClick={handleSubtract}>-1</button>
        <button onClick={handleReset}>Reset</button>
      </div>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};
