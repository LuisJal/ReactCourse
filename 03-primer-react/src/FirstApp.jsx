//Primer componente funcional, debo exportarlo para llevarmelo al main

/*function App(){
    //document.createElemnt
    return <h1>Hola buenas altoque!!</h1>
}

export default App;*/

import React from 'react'

/* export const FirstApp = () => {
  return (
    
  );
};
 */

export const FirstApp=() => (<h1>Hola buenas al toque!!!</h1>)

