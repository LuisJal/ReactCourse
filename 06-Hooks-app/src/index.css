

body {
  padding: 20px;
}

Button{
  margin-right: 10px;
}

.sprite-img {
  width: 150px; /* Ajusta el tamaño según sea necesario */
  height: 150px; /* Ajusta el tamaño según sea necesario */
  background-color: #e4c1c1; /* Fondo clarito */
  border-radius: 50%; /* Esquinas redondeadas */
  margin: 10px; /* Espaciado entre imágenes */
  padding: 10px; /* Espaciado interno */
  object-fit: cover; /* Ajusta la imagen dentro del contenedor */
}