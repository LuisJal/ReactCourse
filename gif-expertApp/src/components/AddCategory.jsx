import React, { useState } from "react";
import {PropTypes} from 'prop-types';

export const AddCategory = ({onNewCategory}) => {
  const [inputValue, setInputValue] = useState();

  const onInputChange = ({ target }) => {
    setInputValue(target.value);
  };

  const onSubmit= (event)=>{
    event.preventDefault();
    if(inputValue.trim().length <= 1) return;
    
    
   // setCategories(categories => [inputValue,...categories]);
    setInputValue('');
    onNewCategory(inputValue.trim());
  }

  return (
    <form onSubmit={(event) => onSubmit(event)} aria-label="form">
      <input
        type="text"
        placeholder="Buscar gifs"
        className="buscador"
        value={inputValue}
        onChange={onInputChange}
      />
    </form>
  );
};


AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired
}