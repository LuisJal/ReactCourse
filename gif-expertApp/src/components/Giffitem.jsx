
import PropTypes from 'prop-types'
export const Giffitem = ({url,tittle,id}) => {
    return (
     <div className="card">
        <img src={url} alt={id} />
        <p>{tittle}</p>

     </div>
        
   
    
  )
}

Giffitem.propTypes ={
   tittle: PropTypes.string.isRequired,
   url: PropTypes.string.isRequired
}
