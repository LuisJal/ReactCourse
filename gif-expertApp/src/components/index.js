//Archivo barril

export * from './AddCategory'
export * from './GiffGrid'
export * from './Giffitem'
