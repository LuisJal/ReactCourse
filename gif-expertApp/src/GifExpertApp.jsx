import { useState } from "react";
import { AddCategory,GiffGrid } from "./components";


export const GifExpertApp = () => {
  const [categories, setCategories] = useState(['Papa']);

  const onAddCategory = (newCategory) => {
    console.log(newCategory);

    //setCategories(['Valorant',...categories]);
    if (categories.includes(newCategory)) return;

    setCategories([newCategory, ...categories]);
  };

  return (
    <>
      <h1>GifExpertApp</h1>
      <AddCategory
        //setCategories = {setCategories}
        onNewCategory={(value) => onAddCategory(value)}
      />

      {categories.map((category) => {
        return <GiffGrid key={category} category={category} />;
      })}
    </>
  );
};

//const apiKey = "kLfb9BxEEvOIHyN9JAlJKke0u9n4hfpB";
