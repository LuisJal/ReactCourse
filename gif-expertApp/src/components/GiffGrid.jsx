import React, { useState, useEffect } from "react";
import { Giffitem } from "./Giffitem";
import { useFetchGifs } from "../hooks/useFetchGifs";
import PropTypes from 'prop-types'

export const GiffGrid = ({ category }) => {
  const { images, isLoading } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>
      {
        isLoading && (<h2>Cargando...</h2>) 
      }
      <div key={category} className="card-grid">
        {images.map((image) => {
          return (
            <>
              <Giffitem key={image.id} {...image} />
            </>
          );
        })}
      </div>
    </>
  );
};

GiffGrid.propTypes = {
  category: PropTypes.string.isRequired
}
