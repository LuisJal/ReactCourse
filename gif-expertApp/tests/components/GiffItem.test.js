import { Giffitem } from "../../src/components";
import { render, screen } from "@testing-library/react";

describe("Pruebas en <GifItem></GifItem>", () => {
  const tittle = "Luis";
  const url = "https://www.luispa.com/";

  test("Debe hacer match con el snapshot", () => {
    const { container } = render(<Giffitem tittle={tittle} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("Debe mostrar la imagen y el url indicados", () => {
    render(<Giffitem tittle={tittle} url={url} />);
    //screen.debug();

    const { src, alt } = screen.getByRole("img");

    expect(src).toBe(url);
    expect(alt).toBe(alt);
  });

  test("Debe mostrar el titulo del componente", () => {
    render(<Giffitem tittle={tittle} url={url} />);
    expect(screen.getByText(tittle)).toBeTruthy();
  });
});
