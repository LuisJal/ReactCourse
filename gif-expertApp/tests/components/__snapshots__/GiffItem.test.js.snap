// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem></GifItem> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      src="https://www.luispa.com/"
    />
    <p>
      Luis
    </p>
  </div>
</div>
`;
